File: CS/WinSolution.Module/Master.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace WinSolution.Module
{
    [DefaultClassOptions]
    public class Master : BaseObject
    {
        public Master(Session session) : base(session) { }
        [Association("Master-Children"), Aggregated]
        public XPCollection<Child> Children {
            get {
                return GetCollection<Child>("Children");
            }
        }
        private string _name;
        public string Name {
            get { return _name; }
            set { SetPropertyValue("Name", ref _name, value); }
        }
    }
}

File: VB/WinSolution.Module/Master.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.Xpo

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace WinSolution.Module
	<DefaultClassOptions> _
	Public Class Master
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		<Association("Master-Children"), Aggregated> _
		Public ReadOnly Property Children() As XPCollection(Of Child)
			Get
				Return GetCollection(Of Child)("Children")
			End Get
		End Property
		Private _name As String
		Public Property Name() As String
			Get
				Return _name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _name, value)
			End Set
		End Property
	End Class
End Namespace
